components/org.wso2.carbon.config/src/main/java/org/wso2/carbon/config/ConfigurationUtils.java

```diff
@@ -109,6 +109,12 @@ public static Properties loadProjectProperties() {
      * @return String substituted string
      */
     public static String substituteVariables(String value) {
+        // Fix the issue #3, This method should not execute in doc generation phase. Need not substitute vaule.
+        // Check the system property to identify the call is coming in doc generation phase.
+        if (Boolean.parseBoolean(System.getProperty(ConfigConstants.SYSTEM_PROPERTY_DOC_GENERATION))) {
```

Comment:
A suggestion from @jsdjayanga to use https://docs.oracle.com/javase/7/docs/api/java/lang/Boolean.html#getBoolean(java.lang.String) instead of the above.